com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -48,62 +35,36 @@ internal enum __RpcExecStage
 #pragma warning restore 414 // restore assigned but its value is never used
 #pragma warning restore IDE1006 // restore naming rule violation check
 
-#pragma warning disable IDE1006 // disable naming rule violation check
-        // RuntimeAccessModifiersILPP will make this `protected`
-        internal NetworkSerializer __beginSendServerRpc(uint rpcMethodId, ServerRpcParams serverRpcParams, RpcDelivery rpcDelivery)
-#pragma warning restore IDE1006 // restore naming rule violation check
-        {
-            PooledNetworkWriter writer;
-
-            SetUpdateStage(ref serverRpcParams.Send);
-
-            if (serverRpcParams.Send.UpdateStage == NetworkUpdateStage.Initialization)
-            {
-                throw new NotSupportedException(
-                    $"{nameof(NetworkUpdateStage.Initialization)} cannot be used as a target for processing RPCs.");
-            }
-
-            var messageQueueContainer = NetworkManager.MessageQueueContainer;
-            var networkDelivery = rpcDelivery == RpcDelivery.Reliable ? NetworkDelivery.ReliableSequenced : NetworkDelivery.UnreliableSequenced;
-
-            if (IsHost)
-            {
-                writer = messageQueueContainer.BeginAddQueueItemToFrame(MessageQueueContainer.MessageType.ServerRpc, Time.realtimeSinceStartup, networkDelivery,
-                    NetworkManager.ServerClientId, null, MessageQueueHistoryFrame.QueueFrameType.Inbound, serverRpcParams.Send.UpdateStage);
-            }
-            else
-            {
-                writer = messageQueueContainer.BeginAddQueueItemToFrame(MessageQueueContainer.MessageType.ServerRpc, Time.realtimeSinceStartup, networkDelivery,
-                    NetworkManager.ServerClientId, null, MessageQueueHistoryFrame.QueueFrameType.Outbound, NetworkUpdateStage.PostLateUpdate);
-
-                writer.WriteByte((byte)MessageQueueContainer.MessageType.ServerRpc);
-                writer.WriteByte((byte)serverRpcParams.Send.UpdateStage); // NetworkUpdateStage
-            }
 
-            writer.WriteUInt64Packed(NetworkObjectId); // NetworkObjectId
-            writer.WriteUInt16Packed(NetworkBehaviourId); // NetworkBehaviourId
-            writer.WriteUInt32Packed(rpcMethodId); // NetworkRpcMethodId
-
-
-            return writer.Serializer;
-        }
-
-#pragma warning disable IDE1006 // disable naming rule violation check
-        // RuntimeAccessModifiersILPP will make this `protected`
-        internal void __endSendServerRpc(NetworkSerializer serializer, uint rpcMethodId, ServerRpcParams serverRpcParams, RpcDelivery rpcDelivery)
-#pragma warning restore IDE1006 // restore naming rule violation check
+        internal void SendServerRpc(ref FastBufferWriter writer, uint rpcMethodId, ServerRpcParams rpcParams, RpcDelivery delivery)
```

Comment:
```suggestion
        // RuntimeAccessModifiersILPP will make this `protected`
        internal void SendServerRpc(ref FastBufferWriter writer, uint rpcMethodId, ServerRpcParams rpcParams, RpcDelivery delivery)
```

Reply:
also one reason we put `__` prefixes to ILPP'd things is to distinguish them from other regular stuff clearly so that we'll always have a clue about what we're playing with.
we didn't have the comment above this method, even if we add it we can still miss that comment because it won't be immediately visible/obvious. (I'll update my suggested change in a sec)

Reply:
Done.